src/main/java/org/jabref/gui/maintable/BibEntryTableViewModel.java

```diff
@@ -68,18 +71,29 @@ public BibEntry getEntry() {
     }
 
     private static ObservableValue<List<AbstractGroup>> createMatchedGroupsBinding(BibDatabaseContext database, BibEntry entry) {
-        Optional<GroupTreeNode> root = database.getMetaData().getGroups();
-        if (root.isPresent()) {
-            return EasyBind.map(entry.getFieldBinding(StandardField.GROUPS), field -> {
-                List<AbstractGroup> groups = root.get().getMatchingGroups(entry)
-                                                 .stream()
-                                                 .map(GroupTreeNode::getGroup)
-                                                 .collect(Collectors.toList());
-                groups.remove(root.get().getGroup());
-                return groups;
-            });
-        }
-        return new SimpleObjectProperty<>(Collections.emptyList());
+        ObservableValue<List<AbstractGroup>> groupsBinding = new ObjectBinding<List<AbstractGroup>>() {
+            {
+                bind(entry.getFieldsObservable(), entry.typeProperty());
```

Comment:
Here we can change the dependence to only be on StandardField.GROUPS again. It could be argued that this change is out of the scope of this PR and has potential performance effects.